apps/hash-api/src/graph/knowledge/system-types/user-secret.ts

```diff
@@ -204,5 +214,35 @@ export const createUserSecret = async <
     },
   );
 
+  const viewPrincipals: PrincipalConstraint[] = [
+    {
+      type: "actor",
+      actorType: "user",
+      id: userAccountId,
+    },
+    {
+      type: "actor",
+      actorType: "machine",
+      id: managingBotAccountId,
+    },
```

Comment:
To make a generic policy we don't need to keep repeating for each secret, I think we need:
1. A condition that if a user's accountId matches the `webId` of the secret, they can see it
2. To somehow be able to determine the `managingBotAccountId`... currently they're not creating the secret, so we'd need to change it so that they are creating it (and therefore we can create a similar rule for the `created_by_id`)

Reply:
It could be the Linear bot, it could also be the Google bot. Basically any current and future bot that might need to access a user secret to use some external API. So I think it has to be somehow tied to a field related to the entity

Reply:
I wonder if it makes sense to introduce some kind of tags here. But this could over-complicate it as well. We can keep it the way we have it for now as it's easy to remove the policies again later.

Reply:
Sounds good